File: credit.c
#include<stdio.h>
#include<cs50.h>
int main(void){
    int length,d1,d2,sum1,sum2,sum3,digit,d;
    long long number,number1;
    printf("Number: ");
    number=get_long_long();
    number1=number;
    length=0;
    d1=0;
    sum1=0;sum2=0;
    while(number1!=0){
        d2=d1;
        d1=number1%10;
        number1=number1/10;
        length++;
        if(length%2==0){
            sum3=0;
            digit=2*d1;
            while(digit!=0){
             sum3=sum3+digit%10;
             digit=digit/10;
            }
            sum1=sum1+sum3;
        }
        else{
            sum2=sum2+d1;
        }
    }
    
    d=d1*10+d2;
    if((sum1+sum2)%10==0){
        switch(d){
            case 51:
            case 52:
            case 53:
            case 54: 
            case 55:
             if(length==16){
            printf("MASTERCARD\n");
            }
            else{
                  printf("INVALID\n");
            }
            
            break;
            case 40:
            case 41:
            case 42:
            case 43:
            case 44:
            case 45:
            case 46:
            case 47:
            case 48:
            case 49:
               if(length==16||length==13){
            printf("VISA\n");
            }
            else{
                  printf("INVALID\n");
            }
            
             break;
             case 34: 
            case 37:
            if(length==15){
            printf("AMEX\n");
            }
            else{
                  printf("INVALID\n");
            }
            break;
            default:
            printf("INVALID\n");
        }
    }
    else{
          printf("INVALID\n");
    }
    
    
}

File: credit.py
number=int(input("Number: "))

number1=number

length=0
d1=0
sum1=0
sum2=0
while True:
    d2=d1
    d1=number1%10
    number1=number1//10
    length=length+1
    if(length%2==0):
        sum3=0
        digit=2*d1
        while True:
            sum3=sum3+digit%10
            digit=digit//10
            if digit==0:
                break
        sum1=sum1+sum3
    else:
        sum2=sum2+d1
    
    if number1==0:
        break

d=d1*10+d2
if (sum1+sum2)%10==0:
    if (d==51 or d==52 or d==53 or d==54 or d==55):
        if length==16:
            print("MASTERCARD")
        else:
            print("INVALID")
       
    if (d==40 or d==41 or d==42 or d==43 or d==44 or d==45 or d==46 or d==47 or d==48 or d==49):
        if length==16 or length==13:
            print("VISA")
        else:
            print("INVALID")
        
    if d==34 or d==37:
        if length==15:
            print("AMEX")
        else:
            print("INVALID")
        
    else:
        print("INVALID")
else:
    print("INVALID")

    
        
    
